### PR TITLE
refactor and new functionality

### PR DESCRIPTION
**Pull request by dpzi**

### CODE REVIEW

**Review by AbhilashZemoso — CHANGES_REQUESTED** [review]
TaskList class should be broken down to multiple classes to achieve Single Responsibility Principle

**Review by AbhilashZemoso — CHANGES_REQUESTED** [review]
use classes and interfaces for all functions

**Comment by dpzi** [comment]
> use classes and interfaces for all functions

JavaScript is module based system I think using class based not good. https://dev.to/giantmachines/stop-using-javascript-classes-33ij

**Comment by AbhilashZemoso** [comment]
> > use classes and interfaces for all functions
> 
> JavaScript is module based system I think using class based not good. https://dev.to/giantmachines/stop-using-javascript-classes-33ij

I agree, but for this assignment we need to implement SOLID principles which can be achieved using interfaces and classes. This is to showcase your understanding in SOLID, where language is just an option.

**Comment by dpzi** [comment]
> > > use classes and interfaces for all functions
> > 
> > 
> > JavaScript is module based system I think using class based not good. https://dev.to/giantmachines/stop-using-javascript-classes-33ij
> 
> I agree, but for this assignment we need to implement SOLID principles which can be achieved using interfaces and classes. This is to showcase your understanding in SOLID, where language is just an option.

No issue i will convert all the files on class

File: typescript/src/task_operations.ts [hunk 1]
```diff
@@ -0,0 +1,176 @@
+import * as util from 'util';
+import * as moment from 'moment';
+import { forEachProject, splitFirstSpace } from "./helper";
+import { println } from "./readlinehelper";
+import { Task, TaskProps } from './task';
+
+
+export class TaskOperations {
```
Comment (AbhilashZemoso) [inline typescript/src/task_operations.ts:8]: can you add a interface for this class, with all the method

Comment (AbhilashZemoso) [reply typescript/src/task_operations.ts:8]: segregate the functions as much as possible in different classes  with individual interfaces

**Review by AbhilashZemoso — APPROVED** [review]
Work on the changes

**Pull request merged**